reana_client/utils.py

```diff
@@ -79,8 +79,16 @@ def load_workflow_spec(workflow_type, workflow_file, **kwargs):
         "serial": serial_load,
         "snakemake": snakemake_load,
     }
-    """Dictionary to extend with new workflow specification loaders."""
-
+    try:
+        """Dictionary to extend with new workflow specification loaders."""
+        load_function = workflow_load[workflow_type]
+    except Exception:
+        logging.warning(
+            'Something went wrong when loading the workflow type: "'
+            + workflow_type
+            + '"'
+        )
+        return
```

Comment:
The output is better but I don't think it's enough to inform the user about what's exactly wrong and what the allowed values are (workflow types).

```console
$ reana-client run                        
[INFO] Creating a workflow...
[WARNING] Something went wrong when loading the workflow type: "foo"
==> Verifying REANA specification file... /Users/marco/code/reanahub/reana-demo-helloworld/reana.yaml
==> ERROR: Cannot create workflow : 
None is not of type 'object'

Failed validating 'type' in schema['properties']['workflow']['properties']['specification']:
    {'id': '/properties/workflow/properties/specification',
     'title': 'Workflow specification in yaml format.',
     'type': 'object'}

On instance['workflow']['specification']:
    None
```

Plus, a different exception might be triggered inside the load function and be caught by this except block,as it's too broad.

Actually, the JSON Schema validation is [not being reached](https://github.com/reanahub/reana-client/blob/3814b05465dc79ad682a14c84e3ae22db1b4eea8/reana_client/utils.py#L189), if you see the error message it's pointing to the `specification` directive, which is `None` where it should be an object.

I think that a better solution would be returning an empty dict spec if the type is not among the valid ones and then we can reach the actual JSON Schema validation.

```diff
diff --git a/reana_client/utils.py b/reana_client/utils.py
index 8be18f7..8ec257b 100644
--- a/reana_client/utils.py
+++ b/reana_client/utils.py
@@ -80,8 +80,10 @@ def load_workflow_spec(workflow_type, workflow_file, **kwargs):
         "snakemake": snakemake_load,
     }
     """Dictionary to extend with new workflow specification loaders."""
-
-    return workflow_load[workflow_type](workflow_file, **kwargs)
+    load_function = workflow_load.get(workflow_type)
+    if load_function:
+        return load_function(workflow_file, **kwargs)
+    return {}
 
 
 def load_reana_spec(
```